internal/webhook/external/github.go

```diff
@@ -197,6 +205,41 @@ func (g *githubWebhookReceiver) getHandler(requestBody []byte) http.HandlerFunc
 			// that URL, we will miss any Warehouses that are subscribed to the same
 			// repository using a different URL format.
 			repoURLs = []string{git.NormalizeURL(e.GetRepo().GetCloneURL())}
+
+		case *gh.RegistryPackageEvent:
+			switch e.GetAction() {
+			// These are the only actions that should refresh Warehouses.
+			case "published", "updated":
+			default:
+				xhttp.WriteResponseJSON(w, http.StatusOK, nil)
+				return
+			}
+			pkg := e.GetRegistryPackage()
+			if pkg == nil || pkg.GetPackageVersion() == nil {
+				xhttp.WriteErrorJSON(
+					w,
+					xhttp.Error(errors.New("invalid request body"), http.StatusBadRequest),
+				)
+				return
+			}
+			switch pkg.GetPackageType() {
+			// These are the only types of packages we care about.
+			case ghcrPackageTypeContainer, ghcrPackageTypeDocker:
+			default:
+				xhttp.WriteResponseJSON(w, http.StatusOK, nil)
+				return
+			}
+			// GitHub sometimes sends package URLs with a trailing colon
+			// and no tag (e.g., "ghcr.io/user/image:"). Such strings are
+			// not valid OCI image references and will result in empty
+			// repo URLs. We trim the trailing colon to avoid parsing errors
```

Comment:
```suggestion
			// GitHub sometimes sends package URLs with a trailing colon and no tag
			// (e.g., "ghcr.io/user/image:"). Such strings are not valid OCI image
			// references. We trim the trailing colon to avoid parsing errors.
```